Code/Framework/AzCore/Tests/DOM/DomPrefixTreeTests.cpp

```diff
@@ -255,37 +255,101 @@ namespace AZ::Dom::Tests
         EXPECT_EQ(80, *tree.ValueAtPath(Path("/foo/0"), PrefixTreeMatch::ExactPath));
     }
 
-    TEST_F(DomPrefixTreeTests, AttachSubTreeAddsNewNode)
+    TEST_F(DomPrefixTreeTests, AttachSubTreeAddsNewNodeAtExistingPath)
     {
         DomPrefixTree<int> tree;
         tree.SetValue(Path("/foo"), 40);
 
         DomPrefixTree<int> treeToAttach;
         treeToAttach.SetValue(Path(), 80);
         treeToAttach.SetValue(Path("/1"), 120);
-        tree.AttachSubTree(Path("/foo/0"), AZStd::move(treeToAttach));
+        bool result = tree.AttachSubTree(Path("/foo/0"), AZStd::move(treeToAttach));
```

Comment:
Doesn't this also mean that you are inserting a new node? If we are going to split the apis as 'attach' and 'attachorinsert', I feel like AttachSubTree should be called ReplaceSubTree and only be allowed to replace nodes in the tree.

Reply:
Actually I feel like using "Attach" and "Insert" together is ambiguous. In the current revision, they can be used to add/replace node.

After some consideration about your suggestion, I would prefer these APIs:
1. **ReplaceNodeAtPath** - Returns true if there is a node that exists already and false if not.
2. **InsertOrReplaceNodeAtPath** - Returns true if it is an insertion and false if it is a replacement. Only insertion adds new node. Replace requires an exact match.
3. **ReplaceSubTree** - same
4. **InsertOrReplaceSubTree** - same

Replace an existing node can also replace any descendants nodes if they exist. For insertion, it is always insert a new node at the exact path that matches. Descendant nodes are either inserted or replaced.